pkg/dataobj/internal/dataset/reader.go

```diff
@@ -394,9 +398,12 @@ func buildMask(full rowRange, s []Row) iter.Seq[rowRange] {
 // Close closes the Reader. Closed Readers can be reused by calling
 // [Reader.Reset].
 func (r *Reader) Close() error {
+	r.region.End()
```

Comment:
This should check to see if region is nil, since if you create a Reader and immediately close it without reading first, region won't be defined and this will panic